meta.go

```diff
@@ -122,6 +122,44 @@ type Storage struct {
 	Properties []string `bson:"properties,omitempty"`
 }
 
+// DeviceType defines a device type
+type DeviceType string
+
+// DeviceGPU represents GPU device type
+const DeviceGPU DeviceType = "gpu"
+
+// DeviceTypes defines current supported device types.
+var DeviceTypes = []DeviceType{DeviceGPU}
+
+// Validate does validation check for DeviceType
+func (dt *DeviceType) Validate() error {
+	for _, t := range DeviceTypes {
+		if *dt == t {
+			return nil
+		}
+	}
+	return fmt.Errorf("Invalid device type: %q, current supported are %+v", dt, DeviceTypes)
+}
+
+// Device represents a charm's device requirement(GPU for example).
+type Device struct {
+	// Name is the name of the device.
+	// name of the GPU device on k8s could be like: nvidia.com/gpu
```

Comment:
This isn't quite right - see the spec. The name is a meaningful name to the charm.
The Type attribute below is where the k8s specific info goes.